src/sgame/sg_buildable.cpp

```diff
@@ -708,6 +710,10 @@ static void HMedistat_Think( gentity_t *self )
 				BG_AddUpgradeToInventory( UP_MEDKIT, client->ps.stats );
 			}
 		}
+		else if ( player->r.svFlags & SVF_BOT )
+		{
+			BotResetStuckTime( player );
```

Comment:
Can we put that in BotActionHeal though? Like after checking that we are healing on medistat or booster speed (`SS_HEALING_…`)

Reply:
Here's an (untested) suggestion, to make implementing this easier.
```diff
diff --git a/src/sgame/sg_bot_ai.cpp b/src/sgame/sg_bot_ai.cpp
index 4947d8f67..a9730b764 100644
--- a/src/sgame/sg_bot_ai.cpp
+++ b/src/sgame/sg_bot_ai.cpp
@@ -1159,6 +1159,31 @@ AINodeStatus_t BotActionFollow( gentity_t *self, AIGenericNode_t *node )
 	return BotMoveToGoal( self ) ? STATUS_RUNNING : STATUS_FAILURE;
 }
 
+// Note: this function assumes that the goal is valid, and is a buildable
+static bool BotCurrentlyHealingWithDedicatedBuildable( gentity_t *self )
+{
+	switch( self->client->ps.stats[ STAT_STATE ] )
+	{
+		case SS_HEALING_2X:
+			// medistation healing if human, good
+			return G_Team( self ) == TEAM_HUMANS;
+		case SS_HEALING_8X:
+			// booster healing if alien, also good
+			return G_Team( self ) == TEAM_ALIENS;
+		case SS_HEALING_4X:
+		{
+			// for human, this is medkit healing, return false.
+			// for aliens, this can be creep healing, in which case
+			// this is good only if we are not aiming towards a booster.
+			return G_Team( self ) == TEAM_ALIENS
+				&& self->botMind->goal.getTargetedEntity()->s.modelindex == BA_A_BOOSTER;
+		}
+		default:
+			return false;
+	}
+}
+
 static AINodeStatus_t BotActionReachHealA( gentity_t *self );
 static AINodeStatus_t BotActionReachHealH( gentity_t *self );
 AINodeStatus_t BotActionHeal( gentity_t *self, AIGenericNode_t *node )
@@ -1197,6 +1222,12 @@ AINodeStatus_t BotActionHeal( gentity_t *self, AIGenericNode_t *node )
 		return STATUS_FAILURE;
 	}
 
+	// If we are properly healing, waiting without moving is good
+	if ( BotCurrentlyHealingWithDedicatedBuildable( self ) )
+	{
+		BotResetStuckTime( self );
+	}
+
 	if ( G_Team( self ) == TEAM_HUMANS )
 	{
 		return BotActionReachHealH( self );
```

I think those `SS_HEALING_[248]X` flags are a huge mess to be honest

Reply:
Testing show that this doesn't work because the `SS_HEALING_4X` and `SS_HEALING_8X` flags are never set for an alien bot. For some reason (bot code executing before buildable code maybe?).

You can ignore this thread honestly, we can fix that later. Or you can integrate this with a comment on the 4X and 8X cases, because it works reliably for humans.

Reply:
This is a better method indeed.